.github/workflows/llvm-ci.yml

```diff
@@ -0,0 +1,263 @@
+name: Build LLVM (matrix)
+
+env:
+  DEFAULT_LLVM_VERSION: '21.1.1'
+
+on:
+  workflow_dispatch:
+    inputs:
+      llvm_version:
+        description: "e.g. 15.0.7, 21.1.1, etc"
+        required: false
+  pull_request:
+    branches:
+      - main
+    paths:
+     - '.github/workflows/llvm-ci.yml'
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    name: ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 180
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-22.04, ubuntu-22.04-arm, macos-15, windows-2025]
+    container: ${{ (matrix.os == 'ubuntu-22.04' && 'quay.io/pypa/manylinux_2_28_x86_64:latest') || (matrix.os == 'ubuntu-22.04-arm' && 'quay.io/pypa/manylinux_2_34_aarch64:latest') || '' }}  
+    defaults:
+      run:
+        shell: bash
+
+    steps:
+      - name: Set env vars
+        id: vars
+        run: |
+          LLVM_VERSION=${{ github.event.inputs.llvm_version || env.DEFAULT_LLVM_VERSION }}
+          echo "LLVM_VERSION=${LLVM_VERSION}" >> $GITHUB_ENV
+          echo "LLVM_TAG=llvmorg-${LLVM_VERSION}" >> $GITHUB_ENV
+
+      # --- calm the git default-branch hint ---
+      - name: Silence git default-branch hint
+        run: git config --global init.defaultBranch main
+
+      # ---------- Prereqs ----------
+      - name: Install prerequisites (Linux)
+        if: runner.os == 'Linux'
+        run: |
+          yum install -y zip
+          export PATH=/opt/python/cp310-cp310/bin:$PATH
+          pip install ninja
+          ln -s /opt/python/cp310-cp310/bin/ninja /usr/bin/ninja-build
+
+      - name: Install prerequisites (Windows)
+        if: runner.os == 'Windows'
+        shell: pwsh
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install ninja cmake
+          git config --global core.autocrlf false
+          Invoke-WebRequest -Uri "https://aka.ms/vs/17/release/vs_BuildTools.exe" -OutFile "vs.exe"
+          .\vs.exe --passive --wait --norestart --includeRecommended --add Microsoft.VisualStudio.Workload.VCTools
+
+      # ---------- Compute install prefix ----------
+      - name: Compute install prefix
+        id: prefix
+        run: |
+          set -euo pipefail
```

Comment:
Why not passing this as part of the `shell` definition, i.e. `shell: bash -ieo pipefail {0}`